### ISSUE TITLE
Array values for attributes are not being properly parsed

### ISSUE THREAD

**Issue by paradox460** [slime-lang/slime#156]
If you supply an array value to an attribute, it is not parsed according to the Slim specification.

Input:
```slim
body class=["hello", "world"]
```

Expected output:
```html
<body class="hello world"></body>
```

Actual output:
```html
<body class="helloworld"></body>
```

---

The array of attributes is not joined properly.

Poking the code revealed this structure as the parsed attributes array:

```elixir
[{"class", {:eex, "\"\#{['hello', ' world']}\""}}]
```

It appears that the attributes are generated into an EEX, which is then evaluated in https://github.com/slime-lang/slime/blob/6a713d4/lib/slime/compiler.ex#L70-L79.

Looking through the parsers and transforms surfaces this likely culprit:

https://github.com/slime-lang/slime/blob/6a713d4/lib/slime/parser/transform.ex#L275-L280
```elixir
def transform(:attribute_value, input, _index) do
  case input do
    {:simple, [_, content, _]} -> to_string(content)
    {:dynamic, content} -> {:eex, to_string(content)}
  end
end
```

Since the attribute_value does _not_ begin with a quote character, the parser flags it as dynamic and the transform encapsulates it in an `eex` structure.

**Comment by Rakoth** [comment]
Thanks for the report, @paradox460! Would you like to make a fix for this in a PR?

**Comment by paradox460** [comment]
I can certainly look into it. I figured a bug would be the best place to start. Do you mind if changes to the basic parsing structure are made, to allow for detecting `[` pairs as attribute values?

**Comment by Rakoth** [comment]
I think we should support not only compile-time lists in attributes, but also lists provided with variables at runtime.
```
Slime.render("""
- var = ["hello", "world"]
body class=var
""")

Slime.render("body class=var", var: ["hello", "world"])
```

**Comment by paradox460** [comment]
I've got a fix open in #157

**Comment by paradox460** [comment]
Hey @Rakoth, bumping this in case it fell through the cracks 👍
